src/components/academy/grading/GradingHistory.tsx

```diff
@@ -23,29 +26,59 @@ class GradingHistory extends React.Component<GradingHistoryProps, {}> {
   }
 
```

Comment:
It is better to split this component into three separate components. Since they are not related to each other, there is no reason to put them here and do a `if/else` chain to select one of them. 

It is also more scalable in future.

Reply:
Agreed; have split the component into three separate components `GradeTooltip`, `GradingStatusTooltip` and `XPTooltip`. Thanks!